src/defabipedia/ABIs_fetcher.py

```diff
@@ -7,29 +7,47 @@
 from defabipedia.types import Chain
 from karpatkit.node import get_node
 
-
-def fetch_abi(contract_address, api_key, chain):
-    if chain.upper() == "POLYGON":
-        url = (
-            f"https://api.polygonscan.com/api?module=contract&action=getabi&address={contract_address}&apikey={api_key}"
+BASE_URL = "https://api.etherscan.io/v2/api"
+
+
+#reference here: https://docs.etherscan.io/etherscan-v2/supported-chains
+
+CHAIN_IDS = {

```

Comment:
also for the numbers of chains I believe we already have that in karpatkit.constants 🤔